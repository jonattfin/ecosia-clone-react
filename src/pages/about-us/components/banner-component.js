import { Button } from '@mui/material';

export default function Component({ language, styles }) {
  return (
    <section className={styles['everyone-power-section']}>
      <div className={styles.title}>We believe in everyone's power to do good</div>
      <div>
        <Button variant="contained">Add Ecosia to Firefox</Button>
      </div>
      <div></div>
    </section>
  )
}
