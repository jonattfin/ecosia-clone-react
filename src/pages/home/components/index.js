import PieComponent from './responsive-pie';

const Components = {
  PieComponent
}

export default Components;
