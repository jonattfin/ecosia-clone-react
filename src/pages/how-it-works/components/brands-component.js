import * as Images from './images';

export default function Component({ styles, language }) {
  return (
    <section className={styles['brands-section']}>
      {getBrands().map((brand, index) => (<img key={`brand_image_${index}`} className={styles.image} src={brand} alt='tdlr'></img>))}
    </section>
  )
}


function getBrands() {
  const { UpWorthyImage, ScientificImage, GuardianImage, SalonImage, ForbesImage } = Images;
  return [UpWorthyImage, ScientificImage, GuardianImage, SalonImage, ForbesImage];
}
