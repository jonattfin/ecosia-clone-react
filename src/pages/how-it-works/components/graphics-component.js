import { Grid } from '@mui/material';

export default function Component({ styles = {}, language }) {
  return (
    <Grid container spacing={2}>
      <Grid item xs={12} xl={12}>
        Over 146 million trees planted
      </Grid>
      <Grid item xs={12} xl={3}>
        <div className={styles.time}>{'December 2009'}</div>
        <div className={styles.subtitle}>{'1.1 sec'}</div>
        <hr className={styles.line} />
        <div className={styles.time}>{'to plant a tree'}</div>
      </Grid>
      <Grid item xs={12} xl={3}>
        <div className={styles.time}>{'over'}</div>
        <div className={styles.subtitle}>{'15 million'}</div>
        <hr className={styles.line} />
        <div className={styles.time}>{'active users'}</div>
      </Grid>
      <Grid item xs={12} xl={3}>
        <div className={styles.time}>{'over'}</div>
        <div className={styles.subtitle}>{'12,101,152'}</div>
        <hr className={styles.line} />
        <div className={styles.time}>{'EUR invested'}</div>
      </Grid>
      <Grid item xs={12} xl={3}>
        <div className={styles.time}>{'over'}</div>
        <div className={styles.subtitle}>{'9000'}</div>
        <hr className={styles.line} />
        <div className={styles.time}>{'planting sites'}</div>
      </Grid>
    </Grid>
  )
}
