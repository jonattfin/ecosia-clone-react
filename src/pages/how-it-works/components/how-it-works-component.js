import * as Images from './images';

import styles from './styles.module.scss';

export default function Component() {
  return (
    <section className={styles['how_it_works_section']}>
      <div className={styles.header}>
        <div className={styles.title}>How it works</div>
      </div>
      <div className={styles.images}>
        {getImages().map((item, index) => (
          <div key={`image_${index}`}>
            <img className={styles.image} src={item.image} alt='tdlr'></img>
            <div className={styles.text}>{item.text}</div>
          </div>
        ))}
      </div>
    </section>
  )
}

function getImages() {
  const { EcosiaImage, AdsImage, IncomeImage } = Images;
  return [
    { image: EcosiaImage, text: 'You search the web with Ecosia.' },
    { image: AdsImage, text: 'Search ads generate income for Ecosia.' },
    { image: IncomeImage, text: 'Ecosia uses this income to plant trees.' },
  ];
}
