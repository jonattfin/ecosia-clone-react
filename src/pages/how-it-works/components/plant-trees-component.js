import { Button } from '@mui/material';

export default function Component({ styles, language }) {
  return (
    <section className={styles['plant-trees-section']}>
      <div className={styles.title}>Plant trees while you search the web</div>
      <div className={styles.separator}>&nbsp;</div>
      <div className={styles['sub-title']}>We use the profit we make from your searches to plant trees where they are needed most. <br />Get the free browser extension and plant trees with every search.</div>
      <div className={styles.separator}>&nbsp;</div>
      <div>
        <Button variant="contained">Add Ecosia to Firefox</Button>
      </div>
    </section>
  )
}
