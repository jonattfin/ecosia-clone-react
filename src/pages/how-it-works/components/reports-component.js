import { Button } from '@mui/material';

export default function Component({ styles, language }) {
  return (
    <section className={styles['reports-section']}>
      <div>
        <img src="https://infopages-prod-cdn.ecosia.org/assets/images/what/transperancy-blue-img-8408403664.svg" alt="transparency" />
      </div>
      <div className={styles.title}>
        We know trust has to be earned
      </div>
      <div className={styles['sub-title']}>
        That is why we publish our monthly financial reports and tree planting receipts. This way you can hold us accountable as we follow our journey to a reforested world.
      </div>
      <div className={styles['view-reports-btn']}>
        <Button variant="contained">View Reports</Button>
      </div>
    </section>
  )
}
