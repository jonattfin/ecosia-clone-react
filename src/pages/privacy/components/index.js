export { default as ProtectPrivacyComponent } from './protect-privacy-component';
export { default as StatementsComponent } from './statements-component';
