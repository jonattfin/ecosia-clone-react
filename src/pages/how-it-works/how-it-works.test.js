import React from 'react';
import renderer from 'react-test-renderer';

import * as Components from './components';

import styles from './styles.module.scss';

const language = {};

it('<Components.BrandsComponent/> renders correctly', () => {
  const tree = renderer
    .create(<Components.BrandsComponent {...{ styles, language }} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('<Components.GraphicsComponent/> renders correctly', () => {
  const tree = renderer
    .create(<Components.GraphicsComponent {...{ styles, language }} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('<Components.HowItWorksComponent/> renders correctly', () => {
  const tree = renderer
    .create(<Components.HowItWorksComponent {...{ styles, language }} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('<Components.PlantTreesComponent/> renders correctly', () => {
  const tree = renderer
    .create(<Components.PlantTreesComponent {...{ styles, language }} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('<Components.ReportsComponent/> renders correctly', () => {
  const tree = renderer
    .create(<Components.ReportsComponent {...{ styles, language }} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
