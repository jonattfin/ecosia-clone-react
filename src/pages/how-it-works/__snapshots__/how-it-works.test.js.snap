// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Components.BrandsComponent/> renders correctly 1`] = `
<section
  className="brands-section"
>
  <img
    alt="tdlr"
    className="image"
    src="up-worthy.svg"
  />
  <img
    alt="tdlr"
    className="image"
    src="scientific.svg"
  />
  <img
    alt="tdlr"
    className="image"
    src="guardian.svg"
  />
  <img
    alt="tdlr"
    className="image"
    src="salon.svg"
  />
  <img
    alt="tdlr"
    className="image"
    src="forbes.svg"
  />
</section>
`;

exports[`<Components.GraphicsComponent/> renders correctly 1`] = `
<section
  className="graphics-section"
>
  <div
    className="title"
  >
    Over 146 million trees planted
  </div>
  <div
    className="column"
  >
    <div
      className="column__item"
    >
      <div
        className="time"
      >
        December 2009
      </div>
      <div
        className="subtitle"
      >
        1.1 sec
      </div>
      <hr
        className="line"
      />
      <div
        className="time"
      >
        to plant a tree
      </div>
    </div>
    <div
      className="column__item"
    >
      <div
        className="time"
      >
        over
      </div>
      <div
        className="subtitle"
      >
        15 million
      </div>
      <hr
        className="line"
      />
      <div
        className="time"
      >
        active users
      </div>
    </div>
    <div
      className="column__item"
    >
      <div
        className="time"
      >
        now
      </div>
      <div
        className="subtitle"
      >
        <div>
          146000000
        </div>
      </div>
      <hr
        className="line"
      />
      <div
        className="time"
      >
        trees total
      </div>
    </div>
    <div
      className="column__item"
    >
      <div
        className="time"
      >
        over
      </div>
      <div
        className="subtitle"
      >
        12,101,152
      </div>
      <hr
        className="line"
      />
      <div
        className="time"
      >
        EUR invested
      </div>
    </div>
    <div
      className="column__item"
    >
      <div
        className="time"
      >
        over
      </div>
      <div
        className="subtitle"
      >
        9000
      </div>
      <hr
        className="line"
      />
      <div
        className="time"
      >
        planting sites
      </div>
    </div>
  </div>
</section>
`;

exports[`<Components.HowItWorksComponent/> renders correctly 1`] = `
<section
  className="how_it_works_section"
>
  <div
    className="header"
  >
    <div
      className="title"
    >
      How it works
    </div>
  </div>
  <div
    className="images"
  >
    <div>
      <img
        alt="tdlr"
        className="image"
        src="ecosia.svg"
      />
      <div
        className="text"
      >
        You search the web with Ecosia.
      </div>
    </div>
    <div>
      <img
        alt="tdlr"
        className="image"
        src="ads.svg"
      />
      <div
        className="text"
      >
        Search ads generate income for Ecosia.
      </div>
    </div>
    <div>
      <img
        alt="tdlr"
        className="image"
        src="income.svg"
      />
      <div
        className="text"
      >
        Ecosia uses this income to plant trees.
      </div>
    </div>
  </div>
</section>
`;

exports[`<Components.PlantTreesComponent/> renders correctly 1`] = `
<section
  className="plant-trees-section"
>
  <div
    className="title"
  >
    Plant trees while you search the web
  </div>
  <div
    className="separator"
  >
     
  </div>
  <div
    className="sub-title"
  >
    We use the profit we make from your searches to plant trees where they are needed most. 
    <br />
    Get the free browser extension and plant trees with every search.
  </div>
  <div
    className="separator"
  >
     
  </div>
  <div>
    <button
      className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      Add Ecosia to Firefox
    </button>
  </div>
</section>
`;

exports[`<Components.ReportsComponent/> renders correctly 1`] = `
<section
  className="reports-section"
>
  <div>
    <img
      alt="transparency"
      src="https://infopages-prod-cdn.ecosia.org/assets/images/what/transperancy-blue-img-8408403664.svg"
    />
  </div>
  <div
    className="title"
  >
    We know trust has to be earned
  </div>
  <div
    className="sub-title"
  >
    That is why we publish our monthly financial reports and tree planting receipts. This way you can hold us accountable as we follow our journey to a reforested world.
  </div>
  <div
    className="view-reports-btn"
  >
    <button
      className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      View Reports
      <span
        className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</section>
`;
